src/network/networkHandler.ts

```diff
@@ -912,7 +872,63 @@ async function handleOnDataMessage(d: OnDataArgs, overworld: Overworld): Promise
         console.error('ADMIN_CHANGE_STAT failed', payload)
       }
       break;
+    }
+  }
+}
+function getFromPlayerViaClientId(clientId: string, underworld: Underworld): Player.IPlayer | undefined {
```

Comment:
What if hotseat and multiple players share the same clientId?

Reply:
This function should only be used in network messages to find out who the message came from, and we had a discord conversation about how this affects hotseat:

> SoulMuncher — 01/17/2024 4:24 PM
Right I was just wondering if theres ever a situation where the fromPlayer could ever be a player not currently taking their turn (maybe some delay in send/recieve messages or some game event or something), in which case this would return the current hotseat player as the "fromPlayer" when that isn't true.
But I don't think there IS a situation where that happens. So this should work fine for finding the fromPlayer. And then if we want to reference another player, that usually gets sent through the payload, so that's taken care of too. 

> JordanO — 01/17/2024 4:32 PM
right, in singleplayer the wspie message gets invoked immediately since there isn't a network to send a message over

It should be fine, since network messages can never come from an inactive player in hotseat. It does feel a little weird, though.

Reply:
Oooh, the globalThis.player check is what catches it for hotseat.  I was looking at the .find and thinking that it wouldn't work for hotseat but it doesn't even get there due to the if check on line 884.  Okay this is good then.